h2/connection.py

```diff
@@ -859,24 +859,40 @@ def send_data(self, stream_id, data, end_stream=False):
         :param end_stream: (optional) Whether this is the last data to be sent
             on the stream. Defaults to ``False``.
         :type end_stream: ``bool``
+        :param pad_length: (optional) Length of the padding to apply to the
+            data frame. Defaults to ``None`` for no use of padding. Note that
+            a value of ``0`` results in padding of length ``0``
+            (with the "padding" flag set on the frame).
+        :type pad_length: ``int``
+            .. versionadded:: 2.6.0
         :returns: Nothing
         """
-        if len(data) > self.local_flow_control_window(stream_id):
+        frame_size = len(data)
+        if pad_length is not None:
+            # Account for padding bytes and the 1-byte padding length field.
+            if pad_length < 0 or pad_length > 255:
+                raise ValueError("pad length must be within range: [0, 255]")
+            frame_size += pad_length + 1
```

Comment:
(No, putting it into the tests doesn’t count. 😉 I might look at the tests if I’ve broken something, but it doesn’t help somebody who’s just casually reading the code.)